testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java

```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
      * block until the rest show up. If there are enough, they will
      * be returned immediately. The returned items will be cleared from the
      * accumulator.
-     * @param desiredQuantity Number of items being awaited
      * @return A list of desiredQuantity many items
      * @throws RuntimeException On failure to attain the requested number of events
      *                          within a reasonable waiting period
      */
     @NonNull
-    public List<HubEvent<?>> take(int desiredQuantity) {
-        // If we haven't yet received the desired quantity of events on the subscription,
-        // setup a latch to await the desired quantity, less the number of existing events.
-        // For example: I desire 5, I already have 3, I wait for 2 more.
-        if (events.size() < desiredQuantity) {
-            latch = new CountDownLatch(desiredQuantity - events.size());
-            Latch.await(latch);
-            latch = null;
-        }
+    public List<HubEvent<?>> takeAll() {
```

Comment:
is there any reason to specify "all" now that it's the only option you provide? I would suggest just `take` but that doesn't sound like its very descriptive of what it's doing. What would be a good verb to follow up on accumulating something? `drain`? Maybe `consume`?

Reply:
I like `drain`. I checked Merriam Webster for antonyms of "accumulate", but none of them felt appropriate here. The "all" was added to indicate the collection return type.

`drain` actually describes the behavior a little better though. This method does drain all items out of the accumulator.